wn/_core.py

```diff
@@ -1546,6 +1547,88 @@ def words(
         """
         return _find_helper(self, Word, find_entries, form, pos)
 
+    @overload
+    def lemmas(
+        self,
+        form: Optional[str] = None,
+        pos: Optional[str] = None,
+        *,
+        data: Literal[False] = False
+    ) -> list[str]: ...
+    @overload
+    def lemmas(
+        self,
+        form: Optional[str] = None,
+        pos: Optional[str] = None,
+        *,
+        data: Literal[True] = True
+    ) -> list[Form]: ...
+
+    # fallback for non-literal bool argument
+    @overload
+    def lemmas(
+        self,
+        form: Optional[str] = None,
+        pos: Optional[str] = None,
+        *,
+        data: bool
+    ) -> Union[list[str], list[Form]]: ...
+
+    def lemmas(
+        self,
+        form: Optional[str] = None,
+        pos: Optional[str] = None,
+        *,
+        data: bool = False
+    ) -> Union[list[str], list[Form]]:
+        """Return the list of lemmas for matching words in this wordnet.
+
+        Without any arguments, this function returns all lemmas in the
+        wordnet's selected lexicons. A *form* argument restricts the
+        words to those matching the given word form, and *pos*
+        restricts words by their part of speech.
```

Comment:
This does not sufficiently explain that only distinct lemma forms are return when `data=False`, rather than "all lemmas in the wordnet's selected lexicons".

Reply:
```
Return the list of lemmas for matching words in this wordnet.

Without any arguments, this function returns all distinct lemma
forms in the wordnet's selected lexicons. A *form* argument
restricts the words to those matching the given word form, and
*pos* restricts words by their part of speech.

If the *data* argument is :python:`False` (the default), only
distinct lemma forms are returned as :class:`str` types. If it
is :python:`True`, :class:`wn.Form` objects are returned for
all matching entries, which may include multiple Form objects
with the same lemma string.
```

I hope this is better.